AutomatticTracks/build.gradle

```diff
@@ -20,23 +20,22 @@ dependencies {
 
     testImplementation 'junit:junit:4.13.2'
     testImplementation 'org.assertj:assertj-core:3.19.0'
-    testImplementation 'org.mockito.kotlin:mockito-kotlin:2.2.11'
-    testImplementation 'org.mockito:mockito-inline:3.8.0'
+    testImplementation 'org.mockito.kotlin:mockito-kotlin:4.0.0'
+    testImplementation 'org.mockito:mockito-inline:4.6.1'
     testImplementation "org.jetbrains.kotlinx:kotlinx-coroutines-test:1.6.4"
 
-    androidTestImplementation 'androidx.test.ext:junit:1.1.2'
-    androidTestImplementation 'androidx.test:runner:1.3.0'
+    androidTestImplementation 'androidx.test.ext:junit:1.1.3'
+    androidTestImplementation 'androidx.test:runner:1.4.0'
 }
 
 android {
-    compileSdkVersion 30
+    compileSdkVersion 33
 
     defaultConfig {
-        versionName "2.1.0"
         minSdkVersion 21
-        //noinspection OldTargetApi
-        targetSdkVersion 30
+        targetSdkVersion 33
```

Comment:
1. Suggestion (💡): Would you consider taking this opportunity and extracting these `minSdkVersion`, `compileSdkVersion` and `targetSdkVersion` into `settings.gradle` so that to make sure that when we next upgrade those we do it once and for all, just like it is already done for WP/WCAndroid?

`settings.gradle`
```
gradle.ext {
    minSdkVersion = 21
    compileSdkVersion = 33
    targetSdkVersion = 33
}
```

`build.gradle`
```
    compileSdkVersion gradle.ext.compileSdkVersion
    defaultConfig {
        minSdkVersion gradle.ext.minSdkVersion
        targetSdkVersion gradle.ext.targetSdkVersion
    }
```

2. Suggestion (💡): On the same spirit, could you please consider doing the same for at least the `test` dependencies that you updated, which was also done multiple times, making it more error prone that it could have been if that was done once and for all. PS: Feel free to do it for `all` the dependencies as well, if that is not to much of a hustle for you.

Reply:
Thanks for leaving those suggestions! I'd opt not to centralize the SDK and/or dependencies versioning. Each module in this project is independent - `AutomatticTracks` and `experimentation` can be integrated into projects without each other, and they don't influence each other. They leave in the same repository just because of pragmatism.

`sampleproject` is just a test app, I don't think we should enforce strictly the same version of SDK or dependencies between them.

It's contrary to WP/WCAndroid apps where modules depend on each other and syncing versions is indeed beneficial. But for Automattic-Tracks-Android, I think that fewer dependencies is a better approach.

Reply:
Thanks for your comment @wzieba ! 🙇 

> I'd opt not to centralize the SDK and/or dependencies versioning. Each module in this project is independent - AutomatticTracks and experimentation can be integrated into projects without each other, and they don't influence each other.

Yes, this totally makes sense, thank you for thinking about that and for deciding not to go with that centralization approach for the reason you wrote. 👍 

However, me looking at this 125f7b704ddea040e3d588ec05f6aa8351ba667c of yours and the fact that all those SDK versions are exactly the same, this begs the question, why not keeping it as centralized, at least for now, and as much as possible, especially if it is already like that. Then, consequent upgrades can continue happening in a centralized manner and only de-centralize in specific and unique situations, when needed, and in isolation? 🤔

> They leave in the same repository just because of pragmatism.

Because I am not fully aware of what the relation is between `AutomatticTracks`, `experimentation` and `sampletracksapp`, and since it seems that you are, then maybe this is a good opportunity to make it clearer for all developers, that is, what is what. So, to my understanding:

`sampletracksapp` is using `AutomatticTracks` and as such those could be related and their SDK centralized.
`experimentation` seems to be an isolated library and not being used in either `AutomatticTracks` or via `sampletracksapp`, thus this can be de-centralized like that. Doing so and making this obvious will help others read this repo a lot faster. 🤔

So, all-in-all I am seeing the issue being `experimentation` and not `sampletracksapp` + `AutomatticTracks` becoming centralized. Correct me if I am wrong. 🙏 

> sampleproject is just a test app, I don't think we should enforce strictly the same version of SDK or dependencies between them.
> 
> It's contrary to WP/WCAndroid apps where modules depend on each other and syncing versions is indeed beneficial. But for Automattic-Tracks-Android, I think that fewer dependencies is a better approach.

Yes, 100%, but then again, look at what happened to `FluxC`, the example app was on `targetSdkVersion = 22` for so long and it now becomes a blocker to even update its `minSdkVersion` (see [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2563)).

I actually consider the sample apps to be on the same boat with WP/WCAndroid and would vote in favor of treating those as such, just so that we are aware the problems with doing such SDK updates on the library level first, via their example apps and then move forward with the SDK update on the clients side. I really find lots of benefits to following such a "centralized" approach.

Obviously, as I tried to explain above, I am also in favor of "de-centralizing" but only when it is really needed (ie `experimentation`. Else, I would recommend to put that bit of an extra effort to upgrade even the example apps, then test those, and only leave them be if they are blocking the upgrade of the individual libraries by a significant amount.

No matter, leaving any library or client or example app behind adds to tech debt that is going to be pained anyway overtime, thus by paying just a little bit of tech debt progressively feels a lot more maintainable to me.

-----

PS: The above is just food for thought, please don't feel the need to action on any of what I am saying to you, apply an changes, yara yara yara... This is just us developers discussing! 😄

Reply:
> this begs the question, why not keeping it as centralized, at least for now, and as much as possible, especially if it is already like that

That's a good point, maybe we can centralize and decentralize when (or _if_) it'll be indeed?

> Because I am not fully aware of what the relation is between AutomatticTracks, experimentation and sampletracksapp, and since it seems that you are, then maybe this is a good opportunity to make it clearer for all developers (...)

That's a good point, I think we should clarify it. This knowledge I have from internal P2s. I see an opportunity for doing this in the scope of #102

Reply:
> That's a good point, maybe we can centralize and decentralize when (or if) it'll be indeed?

Yeap, maybe! 😅 

> That's a good point, I think we should clarify it. The knowledge I have is from internal P2s. I see an opportunity for clarifying relations in the scope of https://github.com/Automattic/Automattic-Tracks-Android/issues/102

Cool, right, this #102 seems a good opportunity to do so indeed! 💯